src/server/app/util/services/inactive-user-email.server.service.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+let _ = require('lodash'),
+	path = require('path'),
+	q = require('q'),
+	deps = require(path.resolve('./src/server/dependencies.js')),
+	handlebars = require('handlebars'),
+	fs = require('fs'),
+	dbs = deps.dbs,
+
+	configc = deps.config,
+	emailService = deps.emailService,
+	logger = deps.logger,
+	User = dbs.admin.model('User');
+
+const day = 86400000;
+
+function buildEmailContent(resource, emailTemplateName) {
+	let numOfDays = Math.floor((Date.now() - resource.lastLogin)/day);
+	let emailData = {
+		appName: configc.app.instanceName,
+		url: configc.app.baseUrl, // message of the day board
+		name: resource.name,
+		date: numOfDays,
+		contactEmail: configc.contactEmail
+	};
+	let emailHTML = fs.readFileSync(`./src/server/app/util/templates/${emailTemplateName}-email.server.view.html`, 'utf-8');
+	let template = handlebars.compile(emailHTML);
+	emailHTML = template(emailData);
+	return emailHTML;
+}
+
+function deactivationAlert(dQuery, config) {
+	return q(User.search(dQuery))
```

Comment:
I would use User.find here, which is a built-in mongoose function that returns everything that matches your query. User.search is a utility we've added to the User model in order to perform paging on searches. We probably won't have enough users that match this query to make a difference, but just in case, we don't want to only return the first "page" of inactive users.

Also, User.find(dQuery).exec() returns a promise already, so you can return it directly instead of wrapping it in q().